### ISSUE TITLE
Enable or disable dark mode via preferences

### ISSUE THREAD

**Issue by molehillrocker** [rancher-sandbox/rancher-desktop#5905]
### Problem Description

At the moment, the dark mode is enabled automatically when the system itself is set to dark mode. Besides of that, I cannot use dark mode in Rancher Desktop while my system doesn't use dark mode.

### Proposed Solution

Add an option to the preferences to allow users to manually enable or disable dark mode (excuse my bad Paint skills 😄):
![image](https://github.com/rancher-sandbox/rancher-desktop/assets/7221361/41708191-00e6-46fc-b3fa-b05529749b11)

**Comment by jandubois** [comment]
This is a good idea. While we could initially put it on the "General" tab, I wonder if we shouldn't create an "Appearance" or "Theme" tag instead, so we can add additional settings in the future, like

* Mode (Light/Dark/Auto)
* Accent Colour
* Font size

I think "Appearance" is the more accurate term, but "Theme" is shorter, so I'm torn.

Assuming we go with "Appearance" then the setting should be string value at the `application.appearance.mode` location with possible values `light`, `dark`, and `auto`, where `auto` is the current behaviour.

In the UI it should be represented with 3 mutually exclusive radio buttons.

**Comment by mook-as** [comment]
Out of curiosity, could you provide more details on why you'd want this? (I recognize that personal preference is typically hard to explain…)  I'm of course not saying we _shouldn't_ do it; just trying to figure out if there's some glaring reason that dark mode (or, possibly an adjustment to our current design) would be beneficial.

This is mostly because I think the text size and colour configuration might be more useful system-wide. (Of course, that also means we _should_ be using system styling but don't…)

**Comment by mook-as** [comment]
> > Out of curiosity, could you provide more details on why you'd want this? (I recognize that personal preference is typically hard to explain…) I'm of course not saying we _shouldn't_ do it; just trying to figure out if there's some glaring reason that dark mode (or, possibly an adjustment to our current design) would be beneficial.
> 
> I guess this is addressed to @molehillrocker? They already answered this, I think with: "I cannot use dark mode in Rancher Desktop while my system doesn't use dark mode."
> 
> So for whatever reason they don't want to use dark mode system-wide, but would want to use it in Rancher Desktop, which seems like a legitimate request.

Yes; I was trying to figure out _why_ that is the case. Is our UI too bright in light mode? I had a phase where Windows was set to use tan for everything, so I can totally understand…

**Comment by molehillrocker** [comment]
> Out of curiosity, could you provide more details on why you'd want this? (I recognize that personal preference is typically hard to explain…) I'm of course not saying we _shouldn't_ do it; just trying to figure out if there's some glaring reason that dark mode (or, possibly an adjustment to our current design) would be beneficial.

Well, first of all it really is personal preference mixed with some applications behaving bad in dark mode when the system is turned into dark mode. Example: An application doesn't specify a background, so the system defaults are used, but for some reason the text color is explicitly set to black, which results in dark text on a dark background.

Another scenario that comes to my mind is that I often work remotely at a customers place, using a Remote Desktop connection. The machine is locked down pretty much, I am not allowed do change anything that is appearance-related, so I can't change backgrounds, adjust fonts or switch to dark mode.

Therefore, I'm used to have my Windows/Linux in whatever style it is (pre)configured and adjust the application-appearance by myself. 😄

**Comment by gaktive** [comment]
We should discuss how this should look so we can tackle this soon.

**Comment by ch0mik** [comment]
up
